dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/guestbook/ManageGuestbooksPage.java

```diff
@@ -234,19 +139,13 @@ private String getFileName() {
     }
 
     public void deleteGuestbook() {
-        if (selectedGuestbook != null) {
-            guestbooks.remove(selectedGuestbook);
-            dataverse.getGuestbooks().remove(selectedGuestbook);
-            try {
-                engineService.submit(new DeleteGuestbookCommand(dvRequestService.getDataverseRequest(), getDataverse(), selectedGuestbook));
-                JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("dataset.manageGuestbooks.message.deleteSuccess"));
-            } catch (CommandException ex) {
-                String failMessage = BundleUtil.getStringFromBundle("dataset.manageGuestbooks.message.deleteFailure");
-                JH.addMessage(FacesMessage.SEVERITY_FATAL, failMessage);
-            }
-        } else {
-            System.out.print("Selected Guestbook is null");
-        }
+        Try.of(() -> manageGuestbooksService.delete(dataverse, selectedGuestbook))
+                .onSuccess(dv -> {
+                    guestbooks.remove(selectedGuestbook);
+                    dataverse.getGuestbooks().remove(selectedGuestbook);
```

Comment:
If we would extract initializing guestbook list from init method then we wouldn't have to worry about removing it from lists and simply run method `refreshGuestbooks(Dataverse)` 
Refresh guestbook method would have everything whats in `init()` afer line `guestbooks = new LinkedList<>();`
We could also split onSuccess to two executions. That is:
```
Try.of(...)
  .onSuccess(this::refreshGuestbooks)
  .onSuccess(JsfHelper.addFlash...)
```